docs/design/gcp-wif-support_design.md

```diff
@@ -0,0 +1,155 @@
+# Design proposal template `<replace with your proposal's title>`
+
+<!-- _Note_: The preferred style for design documents is one sentence per line.
+*Do not wrap lines*.
+This aids in review of the document as changes to a line are not obscured by the reflowing those changes caused and has a side effect of avoiding debate about one or two space after a period.
+
+_Note_: The name of the file should follow the name pattern `<short meaningful words joined by '-'>_design.md`, e.g:
+`listener-design.md`. -->
+
+## Abstract
+Support Google Cloud's WIF (Workload Identity Federation) for OADP.
+
+## Background
+In currently released versions of OADP, the only way to authenticate to GCP is via a service account.
+This is not ideal for customers who are using GCP's WIF feature to authenticate to GCP.
+This proposal aims to add support for WIF to OADP.
+
+## Goals
+- GCP WIF support for OADP and Velero for backup and restore of applications backed by GCP resources.
+- Using OpenShift's Cloud Credentials Operator to generate a short-lived token for authentication to GCP.
+
+## Non Goals
+- ImageStreamTag backup and restore (follow up design/implementation)
```

Comment:
Can you explain, for my edification, why this is a non-goal?

Reply:
I am trying to understand why we should consider these discrete parts of getting this working. IMO getting WIF support means that it works for all the things that we currently get when using the service account.

Reply:
@shawn-hurley *If* we have a use case that matters where local image copy is not required, then getting non-image bits working sooner might be advantageous. On the other hand, if the most important use cases require local image backup, then we probably want it all here.

@shubham-pampattiwar  do you happen to know how important a use case the image backup one is? Is this something relevant to more than 50% of users, or is it more like 10% or lower? It's mainly relevant for use cases where images are manually uploaded to the internal registry -- for Build/BuildConfig-driven images, they'll be rebuilt from scratch on restore of BuildConfig anyway.

Reply:
@sseago To be perfectly I am unsure of the image backup use case relevance number, but maybe, maybe its lets than 50% of the users.

Reply:
Hard for me to tell how many customers find image backup important.  TBH backing up images seems like an anti-pattern and should be discouraged, however the customer is always right :)  To better answer the question, I find it difficult to prioritize image backup.

Reply:
Hard for me to tell how many customers find image backup important.  TBH backing up images seems like an anti-pattern and should be discouraged, however the customer is always right :)  To better answer the question, I find it difficult to prioritize image backup.

Reply:
I still firmly believe we must understand the complete design and the impacts before we decide which parts of work will be available in this release. 

I am not asking that sign up for all the work in the design, just that we think it about and can state what changes we do need for them.

Reply:
So state changes needed to implement non-goals and that we are not doing it yet?
Or move it to goals and mark _to be completed later_?

Reply:
Whichever works for you works for me, I just think that we need to discuss it.

Reply:
Moving into goals as Workload Identity work https://github.com/distribution/distribution/commit/32b33ab6bd6595173762350993093eb2fce9f83b is already in [release-4.14](https://github.com/distribution/distribution/compare/main...openshift:docker-distribution:release-4.14) of docker-distribution so distribution repo work is already eliminated